### PR TITLE
Fix inconsistency about bytes/codepoints by using the number of codepoints to determine how many times tha pad should be pushed.

### PR DESCRIPTION
**Pull request by tormol**
Use the number of codepoints to determine how many times tha pad should be applied.

We can't be _worse_ than the original, can we? ;)

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -68,15 +68,13 @@ pub fn leftpad_with<'a, S>(string: S, len: usize, pad_char: char) -> Cow<'a, str
 /// # Examples
 ///
 /// ```
-/// use left_pad::leftpad;
+/// use left_pad::{leftpad,leftpad_with};
```
Comment (futile) [inline src/lib.rs:78]: Did you actually get an error here? Because the commit says 'Fix doc-test error', and they ran fine for me. Just curious though, I don't really mind moving the imports together.

**Comment by futile** [comment]
Good find! I'm not really familiar with utf-8/codepoints though, so I want to read up on them before I merge this. If you have any links to documentation etc. I'd be thankful for sharing them with me! Might take a few days to get this merged.

File: src/lib.rs [hunk 2]
```diff
@@ -21,32 +21,40 @@ assert_eq!(leftpad(s, 7), "  blubb");
...
 {
     let cow = string.into();
 
-    if !(len > cow.len()) {
+    let cow_codepoints = {
+        let s: &str = cow.borrow();
+        s.len() - s.bytes()
+                   .filter(|b| (b >> 6) == 0b10u8 )
+                   .count()
+    };
```
Comment (futile) [inline src/lib.rs]: Why not use `cow.chars().count()` instead?

File: src/lib.rs [hunk 3]
```diff
@@ -21,32 +21,40 @@ assert_eq!(leftpad(s, 7), "  blubb");
...
 use std::borrow::Borrow;
 
-/// Pads a string to the given length `len` by inserting the character `pad_char` from the left.
+/// Pads a string to the given number of chars by inserting the character `pad_char` from the left.
 ///
-/// If the given string has a length longer or equal to the desired length, it will be
+/// If the given string has more than or exactly the desired number of codepoints, it will be
 /// returned as-is.
 ///
+/// Codepoints are not graphemes, so the result might always not be what a human would expect.
```
Comment (futile) [inline src/lib.rs:29]: Can you add an example (i.e. test, using `assert_eq!`) that shows such a case where the result is different from what "a human" would expect? Also I'm wondering whether it makes sense to take the count of graphemes as the length of the string instead. What are your thoughts on this?

File: src/lib.rs (hunk 1 above)
Comment (tormol) [inline src/lib.rs:78]: yes:

```
   Doc-tests left-pad

running 3 tests
test leftpad_0 ... FAILED
test leftpad_with_0 ... ok
test _0 ... ok

failures:

---- leftpad_0 stdout ----
    <anon>:10:5: 10:32 error: imports are not allowed after non-item statements [E0154]
<anon>:10     use left_pad::leftpad_with;
              ^~~~~~~~~~~~~~~~~~~~~~~~~~~
error: aborting due to previous error
```

File: src/lib.rs (hunk 2 above)
Comment (tormol) [inline src/lib.rs]: I didn't think about it.
`.chars().count()` is probably slower though.

File: src/lib.rs (hunk 3 above)
Comment (tormol) [inline src/lib.rs:29]: I've added an example, but it needs some explanation.
Calculating visual length would require lists of ranges to ignore or count as two.
Filtering everything listed [here](https://en.wikipedia.org/wiki/Combining_character#Unicode_ranges) would be a start, but there are more, like `'ﬃ'` and zero-width chars ([1](https://en.wikipedia.org/wiki/Zero-width_non-joiner), [2](https://en.wikipedia.org/wiki/Zero-width_space))

File: src/lib.rs (hunk 1 above)
Comment (futile) [inline src/lib.rs:78]: Wow, wonder why it worked for me then.. whatevs, I don't really mind whether the imports are combined or not, was just curious.

File: src/lib.rs (hunk 2 above)
Comment (futile) [inline src/lib.rs]: Eh, I don't think it will be _much_ slower, so I'd prefer the `.chars().count()` here for readability.

File: src/lib.rs (hunk 3 above)
Comment (futile) [inline src/lib.rs:29]: Ok, then that should probably not be part of this PR anway. Examples look good, thanks! :)

**Comment by futile** [comment]
Changes look good, as soon as `.chars().count()` is in place I will happily merge :)

**Comment by tormol** [comment]
done

**Comment by futile** [comment]
Great, thanks!
